Manatee.Json/Schema/Keywords/AdditionalPropertiesKeyword.cs

```diff
@@ -128,7 +128,10 @@ public SchemaValidationResults Validate(SchemaValidationContext context)
 				{
 					failedProperties.Add(kvp.Key);
 				}
-				context.EvaluatedPropertyNames.UnionWith(newContext.EvaluatedPropertyNames);
+
+				if (newContext.EvaluatedPropertyNames.Count > context.EvaluatedPropertyNames.Count)
```

Comment:
Will need to do this with `ValidatedIndices` as well.

Reply:
I thought I had. Added it now.